### PR TITLE
Custom amounts UI flow

### LINKED ISSUE #9938: [Custom Amounts M1] Add Custom Amounts flow
The objective of this task is to incorporate the Custom Amounts flow into the order creation process. It's important to note that this ticket does not involve any business logic like adding, removing, or deleting Custom Amounts.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #9938 

### Description

This PR introduces the UI flow for Custom Amounts, allowing navigation from the order creation screen to the custom amounts screen. It's important to note that the business logic is not yet implemented; this PR solely focuses on the necessary UI components and navigation for custom amounts.

### Testing instructions

1. Navigate to the `Orders` tab
2. Click on the "+" button to navigate to the order creation screen.
3. Click on the "Enter Custom Amounts" button in the `products` section
4. Ensure you see a `Custom Amounts` dialog
5. Enter the amount and ensure the numbers only keyboard appears
6. Enter a custom name and ensure the text keyboard appears
7. Press back and ensure the Custom Amounts dialog disappears

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1331230/476dd74a-c89b-46ca-aa3e-cd6301de7a21

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 1]
```diff
@@ -523,6 +522,9 @@ class OrderCreateEditFormFragment :
                     text = getString(R.string.order_creation_add_custom_amounts),
                     onClickListener = {
                         // Implement custom amounts click listener
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]:
```suggestion
```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = ComponentDialog(requireContext(), theme)
+        dialog.onBackPressedDispatcher.addCallback(dialog) {
+            cancelDialog()
+        }
+        return dialog
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt:39]:
```suggestion
        super.onViewCreated(view, savedInstanceState)
```
💡 This line does nothing

File: WooCommerce/src/main/res/layout/dialog_custom_amounts.xml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/major_150"
+                android:gravity="center"
+                android:text="@string/custom_amounts_enter_amount"
+                android:textSize="@dimen/text_major_25"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <com.woocommerce.android.widgets.WCMaterialOutlinedCurrencyEditTextView
```
Comment (samiuelson) [inline WooCommerce/src/main/res/layout/dialog_custom_amounts.xml:67]: 💡 The name input could be improved a bit. I think it would be nice, to restrict it to be single line text, and replace sof "enter" key with "done/next" that would close the keyboard. Let me know what you think.

**Review by samiuelson — APPROVED** [review]
Looks good overall. I've added a few improvement ideas. Let me know what you think.

File: WooCommerce/src/main/res/layout/dialog_custom_amounts.xml (hunk 3 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/res/layout/dialog_custom_amounts.xml:67]: I believe this comment is referring to the `AppCompatEditText` view. In that case, this has been taken care of in the upcoming PRs. Thank you!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt:39]: I see your point. However, I still think keeping it would be a good thing for the below reasons.

**Future Compatibility:** Even if the superclass implementation is empty now, it might not be in future versions of the Android SDK.
**Readability:** Someone reading the code, especially those less familiar with the specifics of Android's internals, might expect to see the super call and could be confused by its absence.
**Extensibility:** If this Fragment is ever subclassed, and that subclass also overrides onViewCreated(), we'd want to be sure the parent's version of the method is called unless we have a specific reason to avoid it.


In summary, while we technically can remove the super call from onViewCreated() without any immediate negative consequences, it's typically a good idea to keep it for the reasons listed above. Let me know what you think. Thank you!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10015/commits/613bffa9c2936662426aa00df371e755aee39656

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt:39]: Feel free to choose your preferred approach. You can find my reasoning below 👇🏼 

* It's misleading to invoke a method that's empty because I assume each line in production code (including the super call) is doing something relevant when I read it. So I find it less readable 😅 

* If we ever want to implement an `open class` (enabling inheritance) and would like to require its descendants to invoke parts of its (`super`) implementation, we should mark these methods with `@CallSuper` annotation, enforcing calling the super implementation explicitly by throwing errors. This is also the current behavior of all the up-to-date Android APIs. We should better not rely on habits or courtesy of API / base class consumers (other programmers).

Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt:39]: Thanks for sharing your opinion. Makes sense! I'll remove this as part of the upcoming PR.
